src/main/java/games/strategy/engine/framework/ArgParser.java

```diff
@@ -18,7 +21,13 @@ public static boolean handleCommandLineArgs(
     if (args.length == 1 && !args[0].contains("=")) {
       // assume a default single arg, convert the format so we can process as normally.
       if (args[0].startsWith(TRIPLEA_PROTOCOL)) {
-        args[0] = GameRunner.TRIPLEA_MAP_DOWNLOAD_PROPERTY + "=" + args[0].substring(TRIPLEA_PROTOCOL.length());
+        final String encoding = StandardCharsets.UTF_8.displayName();
+        try {
+          args[0] = GameRunner.TRIPLEA_MAP_DOWNLOAD_PROPERTY + "="
+              + URLDecoder.decode(args[0].substring(TRIPLEA_PROTOCOL.length()), encoding);
```

Comment:
When I saw this line, my spider sense went off, and I realized that I misspoke before when I said "URL-encoded"; I should have said "URI-escaped" instead.  There is a subtle difference between the two (namely `application/x-www-form-urlencoded` vs. RFC 3986) that is predominantly manifested in how spaces are encoded (`+` vs. `%20`).

This shouldn't affect the use of `URLDecoder#decode()` in this code, though, as it seems to produce the correct decoded/unescaped value for both types of input (see this [SO question](https://stackoverflow.com/questions/607176/java-equivalent-to-javascripts-encodeuricomponent-that-produces-identical-outpu)).  That is, both `+` and `%20` get correctly decoded into a space.

However, this does have implications in the test case below where we use `URLEncoder` to produce a URI-escaped value.  I'll add a separate comment there describing the specific issue.

Tangentially related, but I went over to the website code to see how this value was being generated.  Here's the link element from the map template:

```html
<a href="triplea:{{ page.mapName | escape | replace: " ", "%20"  }}" class="button">Download with TripleA</a>
```

Seeing that we have to do the extra space replacement after running the map name through the `escape` filter, I'm wondering if it would be more appropriate to use the [`url_escape`](https://help.shopify.com/themes/liquid/filters/string-filters#url_escape) filter instead.  That should produce a properly-escaped URI without the extra replacement.

Reply:
https://github.com/triplea-game/triplea-game.github.io/pull/251